jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/LBCSSFolderHideJob.java

```diff
@@ -0,0 +1,95 @@
+import org.sakaiproject.entity.api.ResourcePropertiesEdit;
+import org.sakaiproject.event.api.UsageSessionService;
+import org.sakaiproject.exception.*;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.util.List;
+
+/**
+ * Job to make LB-CSS folder hidden but contents accessible.
+ * LB-CSS folder should not be visible to access user, this job will hide all existing non-hidden LB-CSS folder.
+ * @author neelam
+ *
+ */
+@DisallowConcurrentExecution
+public class LBCSSFolderHideJob implements Job {
+
+	private SqlService sqlService;
+	private SessionManager sessionManager;
+
+	private static final Log LOG = LogFactory.getLog(LBCSSFolderHideJob.class);
+	
+	private ContentHostingService contentHostingService;
+
+	public void setSqlService(SqlService sqlService) {
+		this.sqlService = sqlService;
+	}
+
+	public void setSessionManager(SessionManager sessionManager) {
+		this.sessionManager = sessionManager;
+	}
+
+	public void setContentHostingService(ContentHostingService contentHostingService) {
+		this.contentHostingService = contentHostingService;
+	}
+
+	public void execute(JobExecutionContext context) throws JobExecutionException {
+		ContentCollectionEdit contentCollectionEdit = null;
+		ResourcePropertiesEdit resourceProperties = null;
+		Session session = null;
+		//sql to get collection ids for all LB-CSS folder
+		String sql = "SELECT COLLECTION_ID FROM CONTENT_COLLECTION WHERE COLLECTION_ID LIKE \"%LB-CSS/\" ";
+		LOG.debug("SQL to get collection Id for LB-CSS folder " + sql);
+		int totalFolders = 0 , foldersUpdated = 0, foldersAlreadySet = 0;
+		//set current user in session
+		try {
+			session = sessionManager.getCurrentSession();
+			session.setUserEid("admin");
+			session.setUserId("admin");
+			List<String> collection_ids = sqlService.dbRead(sql);
+			totalFolders = collection_ids.size();
+			LOG.info("Number of LB-CSS folders found is " + totalFolders);
+			for (String id : collection_ids) {
+				try {
+					contentCollectionEdit = contentHostingService.editCollection(id);
+					resourceProperties = contentCollectionEdit.getPropertiesEdit();
+					//if LB-CSS folder is not hidden, hide it
+					if (!("true".equals(resourceProperties.getProperty(ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT)))) {
+						resourceProperties.addProperty(ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT, "true");
+						foldersUpdated++;
+						LOG.info("Hiding the collection" + id +  " from access user ");
+						contentHostingService.commitCollection(contentCollectionEdit);
+					}
+					else {
+						foldersAlreadySet++;
+						LOG.info("The collection " + id +" has already been hidden ");
+					}
+				} catch (SakaiException exception) {
+					LOG.error("Failed to update the LB-CSS folder: " + id, exception);
+				}
+			}
+		}
+		finally {
+			contentHostingService.cancelCollection(contentCollectionEdit);
```

Comment:
I'm not sure about this being here as if there are no ids found it will cause a NullPointerException. I think you probably want it in the else block inside the loop.